# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: /hudson/main/core/src/main/resources/lib/hudson/project
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: matrix.jelly
--- matrix.jelly Base (BASE)
+++ matrix.jelly Locally Modified (Based On LOCAL)
@@ -55,15 +55,21 @@
         <h2>${%Configuration Matrix}</h2>
         <table border="1" class="middle-align center-align">
           <!-- X-axis -->
-          <j:forEach var="x" items="${o.x}">
+          <j:set var="max" value="1"/>
+          <j:forEach var="x" items="${o.x}" varStatus="loop">
             <tr>
               <!-- space for Y-axis -->
               <j:if test="${!empty(o.y)}">
                 <td colspan="${size(o.y)}" />
               </j:if>
-              <j:forEach var="v" items="${x.values}" varStatus="loop">
+              <j:set var="i" value="0"/>
+              <j:while test="${i &lt; max}">
+                <j:forEach var="v" items="${x.values}">
                 <td colspan="${o.width(loop.index)}">${v}</td>
               </j:forEach>
+                <j:set var="i" value="${i+1}"/>
+              </j:while>
+              <j:set var="max" value="${max * x.values.size()}"/>
             </tr>
           </j:forEach>
 
